presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -247,9 +258,14 @@ private synchronized void refreshNodesInternal()
             }
         }
 
-        allNodes = new AllNodes(activeNodesBuilder.build(), inactiveNodesBuilder.build(), shuttingDownNodesBuilder.build());
+        AllNodes allNodes = new AllNodes(activeNodesBuilder.build(), inactiveNodesBuilder.build(), shuttingDownNodesBuilder.build(), coordinatorsBuilder.build());
+        this.allNodes = allNodes;
```

Comment:
Nit: why is this two lines?

Reply:
In " Add abstraction to track minimum worker requirement". I use the `allNodes` local variable in a lambda call back.  This change makes the `allNodes` variable a local variable as opposed to a field which would require a synchronized block to access.  I will move this to that commit and add a comment to make it clear what is going on.